src/FIO/Signer.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Signer.h"
+#include "Address.h"
+#include "../Base58.h"
+#include "../Hash.h"
+#include "../HexCoding.h"
+
+#include <TrezorCrypto/ecdsa.h>
+#include <TrezorCrypto/secp256k1.h>
+
+#include <cassert>
+
+namespace TW::FIO {
+
+using namespace std;
+
+Data Signer::sign(const PrivateKey& privKey, const Data& data) {
+    Data hash = Hash::sha256(data);
+    return signData(privKey, hash);
+}
+
+Data Signer::signData(const PrivateKey& privKey, const Data& data) {
```

Comment:
this looks trivial, can be merged into method `sign`?

Reply:
I made it separate for testing in 2 steps, but now I merge it.